dd-sdk-android-ndk/src/main/cpp/utils/backtrace-handler.cpp

```diff
@@ -56,10 +56,17 @@ namespace {
         return std::string(address_as_hexa);
     }
 
-    void get_info_from_address(size_t index, const uintptr_t address, std::string *backtrace) {
-        backtrace->append(std::to_string(index));
+    void get_info_from_address(size_t index,
+                               size_t start_appending_from_frame_index,
+                               const uintptr_t address,
+                               std::string *backtrace) {
         Dl_info info;
         int fetch_info_success = dladdr(reinterpret_cast<void *>(address), &info);
+        // we still need to unwind the backtrace but we skip adding the frame to the backtrace
+        if (index < start_appending_from_frame_index) {
+            return;
+        }
```

Comment:
Why are you doing this check here and not in the `generate_backtrace` method ?

Reply:
Hmmm I think it was easier here but let me see if I can do this check below.